test/AST-Quake/adjoint-4.cpp

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// RUN: cudaq-quake %s | cudaq-opt | FileCheck %s
+
+#include <cudaq.h>
+
+__qpu__ void init_state(cudaq::qreg<> &qubits, double theta) {}
+
+__qpu__ void reflect_uni(cudaq::qreg<> &ctrls, cudaq::qreg<> &qubits,
+                         double theta) {
+  cudaq::adjoint(init_state, qubits, theta);
```

Comment:
What about controlled? Is that also supported, or were you planning on adding that in a separate PR?

Reply:
There should already be a test for it (it was already implemented). Let me double check though...

Reply:
There are several tests for `cudaq::control` and they have callables of various sorts: plain functions, callable class instances, lambda expressions.